### ISSUE TITLE
Error with version 2.1.2

### ISSUE THREAD

**Issue by marcoschwartz** [VeliovGroup/meteor-autoform-file#45]
Hello,

I am having an issue in my Meteor app since upgrading to 2.1.2. Whenever I open a page that subscribes to the Images collection, I have the error:

[meteor-autoform-files] No such collection "Images" [404]

Before that I had no issues at all. Could this be linked to the latest fix introduced in 2.1.2? Thanks!

**Comment by dr-dimitru** [comment]
Hello @marcoschwartz ,

Show us some of your code:
 1. How collection is defined?
 2. What do you pass into autoform?
 
In general you need to make sure spelling of `Images` collection everywhere is the same starting with capital `I`.

Hope @jankapunkt can provide more help here.

**Comment by jankapunkt** [comment]
Hi @marcoschwartz 

do you use `dburles:mongo-collection-instances` or do you attach the collection on global scope?

@dr-dimitru as I have currently more free time I can provide fixes asap

**Comment by dr-dimitru** [comment]
@marcoschwartz do you want admin access here?

**Comment by marcoschwartz** [comment]
Hi, thanks for the answers! I am not using dburles:mongo-collection-instances, and here I am defining the collection exactly as in the S3 integration example:

https://github.com/VeliovGroup/Meteor-Files/wiki/AWS-S3-Integration

I checked that Images is written the same everywhere.

Then, at the end of the file where I define the collection, I have this:

```
global.Images = Images;

if (Meteor.isServer) {

    Meteor.publish("files.images.all", function(query) {
        return Images.find(query).cursor;
    });

}

export default Images;
```

I am doing something wrong there?

@dr-dimitru Thanks, that would be cool!

**Comment by dr-dimitru** [comment]
@marcoschwartz When I've asked for "How collection is defined?" I mean `new FilesCollection(/*...*/)` constructor.

**Comment by marcoschwartz** [comment]
Sure, here is the whole code for that:

```
var Images = new FilesCollection({

    debug: false,
    collectionName: "Images",
    allowClientCode: true, // Disallow remove files from Client
    onAfterUpload(fileRef) {

        var self = this;

        _.each(fileRef.versions, (vRef, version) => {

            const filePath = "files/" + (Random.id()) + "-" + version + "." + fileRef.extension;

            if (fileRef.ext == 'mp3') {

                var myFuture = new Future();

                mp3Duration(fileRef.path, function(err, duration) {
                    if (err) return console.log(err.message);
                    console.log('Your file is ' + duration + ' seconds long');
                    myFuture.return(duration);
                });

                var duration = myFuture.wait();

                self.collection.update({
                    _id: fileRef._id
                }, { $set: { duration: duration } });

            }

            console.log('Before S3 upload');

            s3.putObject({
                ServerSideEncryption: "AES256",
                StorageClass: "STANDARD",
                Bucket: s3Conf.bucket,
                Key: filePath,
                Body: fs.createReadStream(vRef.path),
                ContentType: vRef.type,
            }, (error) => {

                bound(() => {
                    if (error) {
                        console.error(error);
                    } else {

                        console.log('Uploading to S3');

                        const upd = { $set: {} };
                        upd["$set"]["versions." + version + ".meta.pipePath"] = filePath;

                        self.collection.update({
                            _id: fileRef._id
                        }, upd, function(updError) {
                            if (updError) {
                                console.error(updError);
                            } else {

                                console.log("Uploaded to S3");

                                self.unlink(self.collection.findOne(fileRef._id), version);
                            }
                        });
                    }
                });
            });
        });
    },
    interceptDownload(http, fileRef, version) {

        Meteor.call('insertFileStats', fileRef, http.request.headers);

        let path;

        if (fileRef && fileRef.versions && fileRef.versions[version] && fileRef.versions[version].meta && fileRef.versions[version].meta.pipePath) {
            path = fileRef.versions[version].meta.pipePath;
        }

        if (path) {

            const opts = {
                Bucket: s3Conf.bucket,
                Key: path
            };

            var s3Bucket = s3Conf.bucket;

            // Build S3 path
            if (fileRef.meta.bucket) {
                var s3_path = 'https://s3-' + s3Conf.region + '.amazonaws.com/' + fileRef.meta.bucket + '/' + path;
            } else {
                var s3_path = 'https://s3-' + s3Conf.region + '.amazonaws.com/' + Meteor.settings.s3.bucket + '/' + path;
            }

            Request({
                url: s3_path,
                headers: _.pick(http.request.headers, 'range', 'accept-language', 'accept', 'cache-control', 'pragma', 'connection', 'upgrade-insecure-requests', 'user-agent')
            }).pipe(http.response);

            return true;

        }

        return false;
    }
});
```

**Comment by dr-dimitru** [comment]
@marcoschwartz just to make sure: Collection defined on both Client and Server, right?
All client collection can be found via:
```js
Meteor.connection._mongo_livedata_collections
```

**Comment by jankapunkt** [comment]
@marcoschwartz  I think this is enough code for me to make a reproduction. If it won't reproduce I will upload the repo and post the link here so you can check on that, too.

@dr-dimitru thank you for your trust in me. If you think this makes the process easier than I am good with it. I also have a question regarding tests. I could try to add some tests to the package but it would require to add a test driver package. I am currently going with `meteortesting:mocha`. Would you agree on using this package or would you suggest an alternative?

**Comment by jankapunkt** [comment]
@marcoschwartz I could not recreate your issue so I uploaded a minimal example repo that you can clone and check out.

https://github.com/jankapunkt/autoform-files-issue-45

Note, that I use the gridfs example because it works locally without the need for an S3 account. However, it worked for the autoform also with S3 so you can just replace the code in `Images` to your code and it should run.

As there is the package already hard-copied you can also log some output in the package where the error occurs or set some breakpoints for your debugger.

**Comment by dr-dimitru** [comment]
@jankapunkt We have the same bug report [here](https://github.com/VeliovGroup/Meteor-Files/issues/630#issuecomment-415150806), take a look please.

**Comment by anjunatic** [comment]
My collection code: 
`import { Meteor } from 'meteor/meteor'
// import { Session } from 'meteor/session'
import { FilesCollection } from 'meteor/ostrio:files'

export const ProductImages = new FilesCollection({
	// debug: true,
	collectionName: "ProductImages",
	storagePath() { 
		return (Meteor.settings.publicPath + Meteor.settings.productImagesPath
	)},
	allowClientCode: true, // Required to let you remove uploaded file

	onBeforeUpload(file) {
		// Allow upload files under 20MB, and only in png/jpg/jpeg formats
		if (file.size <= 10485760*2 && /png|gif|jpg|jpeg/i.test(file.ext)) {
			return true
		} else {
			return 'Please upload image, with size equal or less than 20MB'
		}
	}
})

global.ProductImages = ProductImages
`

**Comment by jankapunkt** [comment]
Thank you @anjunatic to confirm the bug. Can one of you provide me a minimal repo that reproduces the error? 

On my last attempt I could not reproduce it, maybe it is a typo hidden somewhere or something bigger that is yet uncovered. A short reply would be great. If the error is discovered I can provide a fix ASAP.

**Comment by anjunatic** [comment]
@jankapunkt here is it! https://github.com/anjunatic/meteor-autoform-files-test

**Comment by jankapunkt** [comment]
Thanks a lot @anjunatic found the cause of this bug and will upload the fix ASAP

**Comment by jankapunkt** [comment]
@marcoschwartz @anjunatic 

I created a fix and pushed it to a new branch [2.1.3](https://github.com/VeliovGroup/meteor-autoform-file/tree/2.1.3)

Please pull and add it as local package and confirm / deny if it fixes the issue. If you need help in adding the package as local package just let me know. Don't forget to checkout the branch before testing

```
cd packages
git clone https://github.com/VeliovGroup/meteor-autoform-file.git
git fetch --all
git checkout 2.1.3
```

In order to support different Meteor versions there are now multiple attempts of retrieving the collection. I will add mocha tests soon in order to make this package more reliable in the long run.

**Comment by dr-dimitru** [comment]
@marcoschwartz @jankapunkt @anjunatic thank you all a lot!
Released as [`v1.2.3`](https://github.com/VeliovGroup/meteor-autoform-file/releases/tag/1.2.3).

Feel free to reopen it in case if the issue is still persists on your end.
